src/fsharp/service/ServiceAnalysis.fs

```diff
@@ -78,56 +80,59 @@ module UnusedOpens =
-                            Range = range
-                            AppliedScope = openDecl.AppliedScope }
-             | _ -> None)
+            if not openDecl.IsOwnNamespace then
+                None
+            else
+                match openDecl.LongId, openDecl.Range with
+                | firstId :: _, Some range ->
+                    if firstId.idText = MangledGlobalName then 
+                        None
+                    else
+                        Some { OpenedGroups = openDecl.Modules |> List.map OpenedModuleGroup.Create
+                               Range = range
+                               AppliedScope = openDecl.AppliedScope }
+                | _ -> None)
 
     /// Only consider symbol uses which are the first part of a long ident, i.e. with no qualifying identifiers
-    let filterSymbolUses (getSourceLineStr: int -> string) (symbolUses: FSharpSymbolUse[]) : FSharpSymbolUse[] =
-        symbolUses
-        |> Array.filter (fun su ->
-             match su.Symbol with
-             | :? FSharpMemberOrFunctionOrValue as fv when fv.IsExtensionMember -> 
-                // Extension members should be taken into account even though they have a prefix (as they do most of the time)
-                true
-
-             | :? FSharpMemberOrFunctionOrValue as fv when not fv.IsModuleValueOrMember -> 
-                // Local values can be ignored
-                false
-
-             | :? FSharpMemberOrFunctionOrValue when su.IsFromDefinition -> 
-                // Value definitions should be ignored
-                false
-
-             | :? FSharpGenericParameter -> 
-                // Generic parameters can be ignored, they never come into scope via 'open'
-                false
-
-             | :? FSharpUnionCase when su.IsFromDefinition -> 
-                false
-
-             | :? FSharpField as field when
-                     field.DeclaringEntity.IsSome && field.DeclaringEntity.Value.IsFSharpRecord ->
-                // Record fields are used in name resolution
-                true
-
-             | :? FSharpField as field when field.IsUnionCaseField ->
-                 false
-
-             | _ ->
-                // For the rest of symbols we pick only those which are the first part of a long ident, because it's they which are
-                // contained in opened namespaces / modules. For example, we pick `IO` from long ident `IO.File.OpenWrite` because
-                // it's `open System` which really brings it into scope.
-                let partialName = QuickParse.GetPartialLongNameEx (getSourceLineStr su.RangeAlternate.StartLine, su.RangeAlternate.EndColumn - 1)
-                List.isEmpty partialName.QualifyingIdents)
+    let filterSymbolUses (getSourceLineStr: int -> string) (checkFileResults: FSharpCheckFileResults) (ct: CancellationToken) : FSharpSymbolUse[] =
+        let filter =
+            fun (su: FSharpSymbolUse) ->
```

Comment:
We never write local functions like this in the compiler.  Actually we generally avoid the use of local functions where they are only used once.  either use `let filter (su: FSharpSymbolUse) = ....` or simply don't use a local function through currying, e.g. 

```
        checkFileResults.GetAllUsesOfAllSymbolsInFileByPredicate ct (fun cu -> ....)
```